components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/main/java/org/wso2/micro/integrator/inbound/endpoint/protocol/grpc/GRPCInjectHandler.java

```diff
@@ -0,0 +1,207 @@
+ */
+public class GRPCInjectHandler {
+    private static final Log log = LogFactory.getLog(GRPCInjectHandler.class);
+    private String injectingSeq;
+    private String onErrorSeq;
+    private boolean sequential;
+    private SynapseEnvironment synapseEnvironment;
+
+    /**
+     * constructor initialize parameters and synapseEnvironment
+     *
+     * @param injectingSeq injecting sequence mentioned in the inbound endpoint
+     * @param onErrorSeq error Sequence mentioned in the inbound endpoint
+     * @param sequential is sequential
+     * @param synapseEnvironment The SynapseEnvironment allows access into the the host SOAP engine. It allows
+     *                           the sending of messages, class loader access etc
+     */
+    GRPCInjectHandler(String injectingSeq, String onErrorSeq, boolean sequential,
+                      SynapseEnvironment synapseEnvironment) {
+        this.injectingSeq = injectingSeq;
+        this.onErrorSeq = onErrorSeq;
+        this.sequential = sequential;
+        this.synapseEnvironment = synapseEnvironment;
+    }
+
+    /**
+     * This method will be called when the initiated gRPC call is expecting an response.
+     * This will inject the gRPC message to the to the message sequence as well as the responseObserver .
+     * inject the message.
+     * @param receivedEvent received event from gRPC
+     * @param responseObserver object to be used when sending the response back to the gRPC client
+     */
+    public void invokeProcess(Event receivedEvent, StreamObserver<Event> responseObserver) {
+        try {
+            org.apache.synapse.MessageContext msgCtx = createMessageContext();
+            if (msgCtx != null) {
+                msgCtx.setProperty(InboundEndpointConstants.INBOUND_ENDPOINT_RESPONSE_WORKER,
+                        new GRPCResponseSender(responseObserver));
+            }
+            initiateSequenceAndInjectPayload(responseObserver, receivedEvent, msgCtx);
+        } catch (SynapseException se) {
+            throw se;
+        } catch (Exception e) {
+            log.error("Error while processing the gRPC Message", e);
+            throw new SynapseException("Error while processing the gRPC Message", e);
+        }
+    }
+
+    /**
+     * This method will be called when the initiated gRPC call is not expecting an response.
+     * This will inject the gRPC message to the to the message sequence as well as the responseObserver .
+     * inject the message.
+     * @param receivedEvent received event from gRPC
+     * @param responseObserver object to be used to send message processing error back to gRPC client
+     */
+    public void invokeConsume(Event receivedEvent, StreamObserver<Empty> responseObserver) {
+        try {
+            initiateSequenceAndInjectPayload(responseObserver, receivedEvent, createMessageContext());
+        } catch (SynapseException se) {
+            throw se;
+        } catch (Exception e) {
```

Comment:
Catch AxisFault only rather than Exception?

Reply:
fixed it in https://github.com/wso2/micro-integrator/pull/854/commits/6fda84c6cf72038734d4872eb4179133b1042af6